pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -18,20 +19,42 @@ const MinSize = 16 // bytes
 // Fingerprint is used to identify a file
 // A file's fingerprint is the first N bytes of the file
 type Fingerprint struct {
-	FirstBytes []byte
+	FirstBytes   []byte
+	HashBytes    uint64
+	BytesUsed    int
+	HashInstance *hash.Hash64
+}
+
+func (f Fingerprint) AddHash() {
+	if f.FirstBytes == nil {
+		return
+	}
+	h := fnv.New64()
+	h.Write(f.FirstBytes)
+	hashed := h.Sum64()
```

Comment:
From a readability standpoint, this allocation seems extraneous to me. Is there a performance benefit to this? I'm asking because this pattern is used quite broadly in this PR.